DESIGN.md

```diff
@@ -33,26 +33,22 @@ During the bootstrap of OpenSearch node, it class loads all the code under `~/pl
 ![](Docs/Extensions.png)
 
 Extensions are independent processes which are built using `opensearch-sdk`. They communicate with OpenSearch via [transport](https://github.com/opensearch-project/OpenSearch/tree/main/modules/transport-netty4) protocol which today is used to communicate between OpenSearch nodes.  
-Extensions are designed to extend features via APIs which are exposed using extension points of OpenSearch.  
+Extensions are designed to extend features via transport APIs which are exposed using extension points of OpenSearch.
 
 ### Discovery
-Extensions are discovered and configured via `extensions.yml` which is read by OpenSearch during the node bootstrap. `ExtensionsOrchestrator` reads through the config file at `~/extensions` and registers extensions within OpenSearch.
+Extensions are discovered and configured via `extensions.yml`, same way we currently have `plugin-descriptor.properties` which is read by OpenSearch during the node bootstrap. `ExtensionsOrchestrator` reads through the config file at `~/extensions` and registers extensions within OpenSearch.
 Here is an example extension configuration `extensions.yml`:
 
 ```
 extensions:
-  - name: opensearch-sdk
-    uniqueId: opensearch-sdk-1
-    hostName: 'sdk_host'
-    hostAddress: '127.0.0.1'
-    port: '4532'
-    version: '1.0'
-    description: Extension for the Opensearch SDK Repo
-    opensearchVersion: '3.0.0'
-    javaVersion: '14'
-    className: ExtensionsRunner
-    customFolderName: opensearch-sdk
-    hasNativeController: false	
+  - name: opensearch-sdk // extension name
+    uniqueId: opensearch-sdk-1 // identifier for the extension
```

Comment:
Shouldn't the name be the unique id?

Reply:
No its not. Unique Id helps when we'd like to run multiple instances of the same extension connected to 1 node of OpenSearch. For example, AD would like to have multiple extensions of itself running to train models etc but still be connected to the same node.

Reply:
Opened up an issue: https://github.com/opensearch-project/opensearch-sdk/issues/65